packages/bsync/src/subscriptions/revenueCatWebhookHandler.ts

```diff
@@ -0,0 +1,174 @@
+import { IncomingMessage, ServerResponse } from 'node:http'
+import AppContext from '../context'
+import { isValidDid } from '../routes/util'
+import { Database } from '..'
+import { sql } from 'kysely'
+import { createSubsOpChannel } from '../db/schema/subs_op'
+
+export const isRevenueCatWebhookUrl = (urlStr: string | undefined) => {
+  if (!urlStr) return false
+  const url = new URL(urlStr, 'http://host')
+  return url.pathname === '/webhooks/revenuecat'
+}
+
+const parseBody = async (req: IncomingMessage) =>
+  new Promise((resolve, reject) => {
+    let body = ''
+    req.on('data', (chunk) => (body += chunk))
+    req.on('end', () => {
+      try {
+        resolve(JSON.parse(body))
+      } catch (err) {
+        reject(err)
+      }
+    })
+    req.on('error', reject)
+  })
+
+// Reference: https://www.revenuecat.com/docs/integrations/webhooks/event-types-and-fields#events-format
+type RevenueCatEventBody = {
+  api_version: '1.0'
+  event: {
+    app_user_id: string
+    type: string
+  }
+}
+
+export const revenueCatWebhookHandler = async (
+  ctx: AppContext,
+  req: IncomingMessage,
+  res: ServerResponse<IncomingMessage>,
+) => {
+  const { db, revenueCatClient } = ctx
+
+  res.setHeader('content-type', 'application/json')
+
+  if (!revenueCatClient) {
+    res.statusCode = 501
+    return res.end(
+      JSON.stringify({
+        error:
+          'Not Implemented: bsync is being served without RevenueCat support',
+      }),
+    )
+  }
```

Comment:
I wonder if we could get in trouble if we hit these error cases and never read/drain the request body.